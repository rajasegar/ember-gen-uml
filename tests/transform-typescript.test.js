const assert = require('assert');
const transform = require('../src/transform-typescript');

describe('transform typescript test: ', function() {
  it('should generate transform public property properly', function() {
    const input = `
export default class MyClass extends Component{
  prop1: boolean = true;
}`;

    const output = `
@startuml


class MyClass {
  +prop1 : Boolean
}

@enduml`;

    const uml = transform('my-class.ts', input, 'MyClass');

    assert.strictEqual(uml, output);
  });

  it('should generate multiple properties properly', function() {
    const input = `
export default class MyClass extends Component{
  prop1: boolean =  true;
  prop2: string  = 'hello';
}`;

    const output = `
@startuml


class MyClass {
  +prop1 : Boolean
  +prop2 : String
}

@enduml`;

    const uml = transform('my-class.ts', input, 'MyClass');

    assert.strictEqual(uml, output);
  });

  it('should generate  public method properly', function() {
    const input = `
export default class MyClass extends Component{
  method1() {
  }
}`;

    const output = `
@startuml


class MyClass {
  +method1()
}

@enduml`;

    const uml = transform('my-class.ts', input, 'MyClass');

    assert.strictEqual(uml, output);
  });

  it('should generate transform service properties  properly', function() {
    const input = `
export default class MyClass extends Component{
  @service i18n!: I18N;
  @service store!: DS.Store;
}`;

    const output = `
@startuml

class i18n << (S, #FF7700) >>
MyClass ..> i18n : service
class store << (S, #FF7700) >>
MyClass ..> store : service
class MyClass {
  
}

@enduml`;

    const uml = transform('my-class.ts', input, 'MyClass');

    assert.strictEqual(uml, output);
  });
});
