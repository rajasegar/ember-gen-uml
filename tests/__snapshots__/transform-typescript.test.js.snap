// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate  public method properly 1`] = `
"
@startuml


class MyClass {
  +method1()
}

@enduml"
`;

exports[`should generate multiple properties properly 1`] = `
"
@startuml


class MyClass {
  +prop1 : Boolean
  +prop2 : String
}

@enduml"
`;

exports[`should generate transform public property properly 1`] = `
"
@startuml


class MyClass {
  +prop1 : Boolean
}

@enduml"
`;

exports[`should generate transform service properties  properly 1`] = `
"
@startuml

class i18n << (S, #FF7700) >>
MyClass ..> i18n : service
class store << (S, #FF7700) >>
MyClass ..> store : service
class MyClass {
  
}

@enduml"
`;
